manifests/init.pp

```diff
@@ -28,6 +28,7 @@
   Optional[Enum['rsa', 'prime256v1', 'secp384r1']] $key_algo             = undef,
   String                                           $contact_email, # lint:ignore:parameter_order
   Optional[Boolean]                                $ocsp_must_staple     = undef,
+  Optional[Integer[0]]                             $timeout              = 300,
```

Comment:
[300 is the default value](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/type/exec.rb#L287-L304). If you want the module to not set a particular value unless the user sets one, I guess defaulting to `undef` would be fine?

Also, please "document" this new setting into the *Reference* section of `README.md`.